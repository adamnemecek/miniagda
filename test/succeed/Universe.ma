-- 2010-08-28

data Nat : Set
{ zero : Nat
; suc  : Nat -> Nat
}

mutual {

  data U : Set 
  { nat : U
  ; pi  : (a : U) -> (El a -> U) -> U
  }

  fun El : U -> Set
  { El nat = Nat
  ; El (pi a f) = (x : El a) -> El (f x)
  }
}