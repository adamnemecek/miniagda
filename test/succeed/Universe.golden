MiniAgda by Andreas Abel and Karl Mehltretter
--- opening "Universe.ma" ---
--- scope checking ---
--- type checking ---
type  Nat : Set
term  Nat.zero : < Nat.zero : Nat >
term  Nat.suc : ^(y0 : Nat) -> < Nat.suc y0 : Nat >
type  U : Set
term  U.nat : < U.nat : U >
term  U.pi : ^(a : U) -> ^(y1 : El a -> U) -> < U.pi a y1 : U >
{ El U.nat = Nat
; El (U.pi a f) = (x : El a) -> El (f x)
}
--- evaluating ---
--- closing "Universe.ma" ---
